E2E-ORS-Sample-For-Java/src/main/java/ObjectReplicationMonitor.java

```diff
@@ -0,0 +1,297 @@
+
+        // Get the container client to access blob client
+        BlobContainerClient blobContainerClient = blobServiceClient.getBlobContainerClient(containerName);
+
+        // Variables to keep track of and represent how many blobs have replicated
+        int completed = 0;
+        double ratio = 0;
+        int percentage = (int) (ratio * 100);
+        int size = blobList.length;
+        ArrayList<String> completedEvents = new ArrayList<>();
+        ArrayList<String> failedEvents = new ArrayList<>();
+
+        // While loop for continuous updating on replication status until all blobs have replicated
+        while (completed < size) {
+
+            // Iterate through list of all blobs
+            for (String b : blobList) {
+
+                // Check if event has already been completed, as to only check status of unreplicated blobs
+                if (!completedEvents.contains(b) && !failedEvents.contains(b)) {
+
+                    // Get blob client to access blob
+                    BlobClient blobClient = blobContainerClient.getBlobClient(b);
+
+                    // Get properties of blob
+                    BlobProperties properties = blobClient.getProperties();
+                    List<ObjectReplicationPolicy> propertyList = properties.getObjectReplicationSourcePolicies();
+
+                    // Find replication policy status
+                    for (ObjectReplicationPolicy policy : propertyList) {
+                        for (ObjectReplicationRule rule : policy.getRules()) {
+
+                            // Track how many replications have completed
+                            String status = rule.getStatus().toString();
+
+                            // If the replication was completed, add 1 to variable completed and add that blob to the failedEvents list
+                            if (status.equals("complete")) {
+                                completed++;
+                                completedEvents.add(b);
+                            }
+
+                            // If the replication was failed on a specific blob, alert user and continue checking replication of other blobs
+                            else if (status.equals("failed")) {
+                                System.out.println("\nReplication of " + b + " has failed. Please check that " + b + " in destination container are not archived...");
+                                failedEvents.add(b);
+                                size--;
+                            }
+                        }
+                    }
+                }
+
+                // Check if status has a new value and percentage, to avoid repeat printing the same status
+                if ((double) completed / (double) size != ratio && (int) (((double) completed / (double) size) * 100) != percentage) {
+                    ratio = (double) completed / (double) size;
+                    percentage = (int) (ratio * 100);
```

Comment:
It might look a bit more clear if you pulled completed/size out into its own variable so something like: 
double status = (double)completed / (double) size;
if (status != ratio && (int)(status * 100) != percentage) {
ratio = status; ...
}